File: dotnet/ThirtyFour.UI.Tests/DriverFactory.cs
using System;
using System.Collections.Generic;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using OpenQA.Selenium.Firefox;

namespace ThirtyFour.UI.Tests
{
    static class DriverFactory
    {
        public static IWebDriver BuildDriver(IDictionary<string, object> properties)
        {
            IWebDriver driverInstance;

            var browser = properties["browser"];

            switch (browser.ToString().ToLower())
            {
                case "chrome":
                    // chromedriver in path, or set ....
                    // System.Environment.SetEnvironmentVariable("webdriver.chrome.driver", @"path/to/chromedriver");

                    var chromeService = ChromeDriverService.CreateDefaultService();
                    var chromeOptions = new ChromeOptions();
                    driverInstance = new ChromeDriver(chromeService, chromeOptions);
                    break;
                case "firefox":
                    // geckodriver in path, or set ....
                    // System.Environment.SetEnvironmentVariable("webdriver.gecko.driver", @"path/to/geckodriver");

                    var firefoxService = FirefoxDriverService.CreateDefaultService();
                    var firefoxOptions = new FirefoxOptions();
                    driverInstance = new FirefoxDriver(firefoxService, firefoxOptions);
                    break;
                default:
                    throw new ArgumentException($"Invalid local browser parameter - '{browser}'");
            }

            driverInstance.Manage().Window.Maximize();
            driverInstance.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(Double.Parse(properties["implicitWaitSeconds"].ToString()));
            return driverInstance;
        }

       
    }
}

File: java/src/test/java/com/github/danhumphrey/thirtyfour/ui/test/DriverFactory.java
package com.github.danhumphrey.thirtyfour.ui.test;

import java.net.MalformedURLException;
import java.util.concurrent.TimeUnit;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.firefox.FirefoxOptions;

public class DriverFactory
{
	
    public static WebDriver buildDriver() throws MalformedURLException
    {
        WebDriver driver = null;

        switch (TestProperties.BROWSER)
        {
        	case "${browser}": // resource filtering not applied
        		driver = new ChromeDriver();
                break;
            case "chrome":
            	ChromeOptions chromeOptions = new ChromeOptions();
            	if(TestProperties.HEADLESS) {
            		chromeOptions.addArguments("headless");
            	}
                driver = new ChromeDriver(chromeOptions);
                break;
            case "firefox":
            	FirefoxOptions firefoxOptions = new FirefoxOptions();
            	if(TestProperties.HEADLESS) {
            		firefoxOptions.addArguments("--headless");
            	}
                driver = new FirefoxDriver(firefoxOptions);
                break;
            default:
                throw new IllegalArgumentException(String.format("Invalid browser - '%s'", TestProperties.BROWSER));
        }
        driver.manage().window().maximize();
        driver.manage().timeouts().implicitlyWait(2, TimeUnit.SECONDS);
        return driver;
    }
    
  
 
}